pkg/metrics/metrics.go

```diff
@@ -325,12 +187,15 @@ func (c *Config) ensureServiceMonitor() error {
 			},
 		},
 		Spec: monv1.ServiceMonitorSpec{
+			AttachMetadata: &monv1.AttachMetadata{
+				Node: &attachMetadataBool,
+			},
 			Endpoints: []monv1.Endpoint{
 				{
 					HonorLabels:     true,
 					Interval:        "30s",
 					Path:            "/metrics",
-					Port:            "metrics",
+					Port:            "https-metrics",
```

Comment:
Does this have a functionality change?

Reply:
no but this needs to be same as the kubelet service port